proposed/2021/NuGet.orgTFMs.md

```diff
@@ -0,0 +1,96 @@
+# NuGet Target Frameworks on Package Details Page 
+
+- Author Name (https://github.com/jcjiang)
+- Start Date (2021-02-16)
+- GitHub Issue (https://github.com/NuGet/NuGetGallery/issues/8387)
+- GitHub PR ()
+
+## Summary
+
+When a developer opens the package details page or tab on NuGet.org or NuGet rooling, they will see what frameworks the package can support. 
+
+## Motivation
+
+### Problem: What problem is this solving? 
+
+Developers cannot use the package details page to determine what frameworks a NuGet package can support. In 2021, we live in a world where there are three evolutions of .NET coexisting on NuGet; .NET Framework, .NET Core, and .NET 5.  
+
+Developers should be able to determine whether a package works for their project without having to first download it or rely on the author’s documentation. 
+
+### Why: How do we know this is a real problem and worth solving? 
+
+Finding a package that is compatible with your project is a challenge for the average developer in the .NET ecosystem. Over 21% of developers fail to install a package today. One of the top NuGet errors issued in telemetry is [NU1202](https://docs.microsoft.com/en-us/nuget/reference/errors-and-warnings/nu1202), which shows that we allow developers to install packages that are not compatible with their project or solution’s target framework. 
+
+### Success: How do we know if we’ve solved this problem? 
+
+- By measuring the change in the percentage of successful installs and lowering the amount of [NU1202](https://docs.microsoft.com/en-us/nuget/reference/errors-and-warnings/nu1202) errors in telemetry. 
+- By measuring the usage of target framework & supported platform filters on NuGet.org & Visual Studio. 
+- By gauging excitement/disappointment after blogging about this feature & analyzing sentiment on Twitter/GitHub/DevCom/etc. 
+
+### Audience: Who are we building for? 
+
+We are building this for the .NET developer who is browsing & making decisions about including a NuGet package into their project based on target framework & supported platform criteria. 
+
+## Explanation
+
+### Functional Explanation
+
+#### Context and Scope 
+When a developer accesses the details page for a NuGet package, they will see information on target frameworks in a dropdown menu. The framework information will be surfaced from the package. In cases where versions are not specified (such as .net) or other unexpected scenarios, ‘Version Not Specified’ or similar messages will be surfaced. 
+
+##### Minimal Requirements 
+- List of Frameworks on NuGet.org
```

Comment:
- What to do with non-Dependency type packages? e.g. .NET tools?
- What about `tools` assets in dependency packages? will `tools/net40` make the package appear as compatible with net40?

Reply:
_Do_ we consider `tools/net40` to be compatible with `net40`?

Reply:
Yep. `tools/{TFM}` would represent that the tool is supported on that version of .NET. We should list it the same way in the sense that a tool can be compatible with net5.0 / netcoreapp3.1 / etc.

Reply:
Added `- Runtime, Content Build assets and Tools packages will display TFM details.`

Reply:
`tools/{tfm}/{rid}/{any}` is not used for dependency package compatibility ([source](https://github.com/NuGet/NuGet.Client/blob/97311b5cb48d38402b49fd41a6384b11db7dd453/src/NuGet.Core/NuGet.Commands/RestoreCommand/CompatibilityChecker.cs#L431-L443)). If we choose to display it, people will still encounter failed installs with based on this information.

Reply:
Ah, that's interesting. Yes, that's the risk of including a pattern set that's not called out in the restore code's compat checks. This was a tenuous inclusion anyway, and I can remove it if we don't want this kind of misleading. Build assets could give us the same problem though, right?

Reply:
Build assets are considered in the compat check:
https://github.com/NuGet/NuGet.Client/blob/97311b5cb48d38402b49fd41a6384b11db7dd453/src/NuGet.Core/NuGet.Commands/RestoreCommand/CompatibilityChecker.cs#L441-L442

Reply:
Not the same set, by the looks: https://github.com/NuGet/NuGetGallery/blob/dg-tfmbackfill/src/GalleryTools/Commands/BackfillTfmMetadataCommand.cs#L65-L67
For a purer experience, perhaps I should remove transitive build pattern sets.

Returning to tools, if we present a TFM based on the tools pattern set, and there is no compat checking on restore, isn't it more likely that the _NuGet.org_ experience is more restrictive? In other words, would we see TFMs on some tools packages that would always restore, but not all restorable tools packages would have TFMs on NuGet.org?

Reply:
After discussion offline, we're looking to:
- keep build asset patterns but drop out transitive build assets in order to make this compat check consistent with nuget restore
- only assess tools pattern sets in the case of tools packages (also consistent with restore logic).

This is beyond the scope of a UX spec, of course, but relevant in making it happen.